boromir674/cookiecutter-python-package
# Cookiecutter Python Package v1.9.0 Release

1.9.0 (2023-12-22)
==================

Changes
-------

feature
"""""""
- feature CI/CD Pipeline and Ruff in Generated README's Dev Section
- automatically run Ruff on Generated Project CI Pipeline
- make Generator, and Template Pass Ruff Checks/Evaluations
- add Ruff to Template Proj, as available tox command for fast Static Code Check

fix
"""
- remove extra but empty Log file, that appears inside the Gen Project

test
""""
- automatically test that Rendered policy_lint.yml CI config is valid YAML
- automatically verify that extra, but empty log file does NOT appear in Gen Proj
- update Biskotaki Test Snapshot, produced in Interactive Mode

documentation
"""""""""""""
- feature Ruff as New Fast Python Linter, and add Ruff Code Badge in README
- mention pytest-explicit as requirement for all test suite features (ie cli flags)

style
"""""
- apply isort and black code "fixes"

refactor
""""""""
- apply Ruff fixes
- apply black to code

ci
""
- set CI/CD Production Settings
- run only Lint Test on CI and Force Lint
- run Ruff in the Static Code Analysis CI Job and keep running legacy Pylint in separate Job
- fix reference of output Variable
- set CI/CD Production Settings
- remove dedicated docker settings reading Job
- refactor so that some intermediate Jobs are eliminated
- delegate Code Visualization to Reusable Workflow, with configurable execution Policy
- use workflow one-liner instead of bash if-else
- account for gh failing to read from env
- add quotes arroudn expression
- set CI pipe settings to ONLY Lint
- use Lint Job from boromir674/automated-workflows

chore
"""""
- add 'Snapshot Update' Scripts, and dedicate README for the Process

release
"""""""
- bump version to 1.9.0

tox
"""
- clean code
